pkg/http/middleware/auth.go

```diff
@@ -0,0 +1,61 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"go.signoz.io/signoz/pkg/types/authtypes"
+	"go.uber.org/zap"
+)
+
+type Auth struct {
+	logger *zap.Logger
+	jwt    *authtypes.JWT
+}
+
+func NewAuth(logger *zap.Logger, jwt *authtypes.JWT) *Auth {
+	if logger == nil {
+		panic("cannot build auth middleware, logger is empty")
+	}
+
+	return &Auth{logger: logger, jwt: jwt}
+}
+
+func (a *Auth) Wrap(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// if there is a PAT token, attach it to context
+		patToken := r.Header.Get("SIGNOZ-API-KEY")
+		if patToken != "" {
+			ctx := context.WithValue(r.Context(), authtypes.PatTokenKey, patToken)
+			r = r.WithContext(ctx)
+		}
```

Comment:
```suggestion
		// if there is a PAT token, attach it to context
		patToken := r.Header.Get("SIGNOZ-API-KEY")
		if patToken != "" {
			ctx := context.WithValue(r.Context(), authtypes.PatTokenKey, patToken)
			r = r.WithContext(ctx)
		}
```
This cannot come in the OSS offering. We need to create another middleware in ee codebase.

Reply:
this just add's the PAT to the context, the actual parsing and everything happens the auth middleware present in EE code. Will you still suggest to move this there ?